sdk/java/pulumi/src/main/java/com/pulumi/deployment/internal/DeploymentInstanceInternal.java

```diff
@@ -19,17 +20,21 @@
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Function;
 
+import static java.util.Objects.requireNonNull;
+
 /**
  * Metadata of the deployment that is currently running. Accessible via @see {@link com.pulumi.deployment.Deployment#getInstance()}.
  */
 @InternalUse
 public final class DeploymentInstanceInternal implements DeploymentInstance {
```

Comment:
moving `ConfigInternal` here gives us less tight coupling and makes future development easier and the implementations is made simpler

IMHO the `DeploymentInstanceInternal` should be renamed `GlobalContext` and separated from the `Deployment` in future refactoring to clearly state its role in the architecture, but this is just a remark and remains out of scope for this PR.